js/src/helpers/taxonomy-replacevars.js

```diff
@@ -8,86 +8,45 @@ import {
 } from "@wordpress/data";
 import forEach from "lodash/forEach";
 
+/* Internal dependencies */
+import {
+	promisifySelector,
+	invalidateTerms,
+} from "../wp-data/utils";
+
 /**
  * Creates a promise that will resolve to the post's post type.
  *
  * @returns {Promise<string>} A promise resolving to the post type.
  */
-function getPostType() {
-	return new Promise( resolve => {
-		let postType = select( "core/editor" ).getEditedPostAttribute( "type" );
-
-		if ( postType ) {
-			return resolve( postType );
-		}
-
-		const unsubscribe = subscribe( () => {
-			postType = select( "core/editor" ).getEditedPostAttribute( "type" );
-
-			if ( postType ) {
-				unsubscribe();
-
-				return resolve( postType );
-			}
-		} );
-	} );
-}
+const getPostType = promisifySelector(
+	"core/editor",
+	"getEditedPostAttribute",
+).bind( null, "type" );
 
 /**
- * Creates a promise that will wait for getEntityRecords to yield a result with the given arguments.
+ * Creates a promise that will resolve to a list of taxonomies.
  *
- * @param {Array} args The arguments to be passed to getEntityRecords.
+ * @param {string} posttype The post type to get taxonomies for.
  *
- * @returns {Promise} A promise resolving to the entity records.
+ * @returns {Promise<array>} List of taxonomies for the given post type.
```

Comment:
After your explanation, this returns a Promise of an object right? The current state of `taxonomies`, filtered by post type.